### ISSUE TITLE
Notebooks

### ISSUE THREAD

**Issue by albermax** [albermax/innvestigate#18]
We use (ipy)notebooks as main way to introduce people to the api. "Learning by doing".

The following notebooks and contents should be part of the first release:

* A) [core reversal code:] walk through the idea of inverting the graph, show how to implement the gradient by inverting each layer, then a slightly more advanced example by showing how to create deconvnet and guided backprop.
  * best done with inception v3 => fast network and imagenet pics look great.
  * If there are some nice visualization/sketches on how the reversion works, we can use them also in the paper. Parts of the text can also be shared.

* B) [mnist complete workflow:] the same as the mnist example right now, just with more comments on how things work. And the code should be structured more nicely, i.e, not using those function, more in a step by step fashion.
  * the pics at the moment look horrible, I think this is due to the training. Best look into the cat-paper and try to reproduce the results, in case ask Sebastian or Pieter-Jan how the got them.

* C) [imagenet application] a step-by-step example with imagenet, again similar to the all_methods.py. The scope here is to reuse the patterns and let the people know how the can use our applications submodule.

* D) [imagenet network comparison] Same as the C only the outcome should be a square with n networks as rows and n methods as columns. The code can and will be a bit more messy as different networks need different preprocessing functions and image sizes. Overall it should be doable as the key information is already present in the dictionary returned by innvestigate.applications.
  * Reference C and delimit the scope to show the power of innvestigate and on how to compare different networks.
  * The plots of this example will be used in the final paper.

* E) [imagent pattern training] similar to the current train_***.py. Focus on how to train patterns for a large network. In the best case this example shows how to train with several gpus (means setting the parameter gpus=X).

* D) [perturbation mnist] an example notebook on how to use perturbation analysis with mnist that everybody can run.

* E) [perturbation imagenet] an example notebook on how to use perturbation analysis with imagenet.

* F) [LRP/DT intro] Sebastian might wants to add an LRP/DT notebook.

I think if there exists first a working python-script it is easy to create the notebook. The advantage would be one can run the examples easily via ssh. The drawback is the code is doubled and when changed one needs to change two places.
